src/fun_lti_provider/default/backend.py

```diff
@@ -0,0 +1,197 @@
+"""This module contains an authentication backend based on LTI launch request"""
+
+import logging
+from typing import Any, Optional
+
+from django.contrib.auth import get_user_model
+from django.core.exceptions import PermissionDenied
+from django.http import HttpRequest
+
+from fun_lti_provider.lti import LTI
+
+logger = logging.getLogger("fun_lti_provider")
+USER_MODEL = get_user_model()
+
+
+class LTIBackend:
+    """
+    Authentication backend used by fun_lti_provider.default.handlers.success handler
+    It authenticates an user from a verified LTI request and creates an User if necessary.
+
+    You are encouraged to make your own authentication backend, by overriding this one to
+    add your own domain logic.
+    """
+
+    def authenticate(self, request: Optional[HttpRequest], **kwargs):
+        """
+        Authenticate an user from a LTI request
+
+        Args:
+            request: django http request
+            kwargs: additional parameters
+
+        Returns:
+            An authenticated user or None
+        """
+
+        lti_request = kwargs.get("lti_request")
+        if not lti_request:
+            return None
+
+        if not self.check_lti_request_content(lti_request):
+            return None
+
+        username = self.get_username_from_lti_request(lti_request)
+        logger.debug("User %s authenticated from LTI request", username)
+
+        try:
+            user = USER_MODEL.objects.get(username=username)
```

Comment:
should we take care of the `is_active` flag to be able to deactivate a user and forbid access even via LTI?

Reply:
You're right! I updated `authenticate()` and `get_user()` to take care of the `is_active` flag.